test/unit/lib/request-with-oauth2.js

```diff
@@ -53,72 +53,58 @@ describe('lib/requestWithOAuth2', function() {
-      requestGet.onFirstCall().yields(null, { statusCode: 200 }, {});
+    it('GETs the configured url when no OAuth2 credentials are provided', function (done) {
+      requestGet.onFirstCall().yields(null, {statusCode: 200}, {});
 
       requestWithOAuth2.get('http://testhost.com', undefined, undefined, undefined, undefined, undefined,
-        function(err) {
-          if (!err) {
-            request.get.should.have.been.calledWith('http://testhost.com');
-          }
-          done(err);
+        function (err, res, body) {
+          request.get.should.have.been.calledWith('http://testhost.com');
+          done()
         });
     });
 
     it('GETs an authorization token from the oAuth2 server and uses returned auth token in GET request to URL',
-      function(done) {
-        requestPost.onFirstCall().yields(null, { statusCode: 200 }, JSON.stringify(successfulAuthResponse));
-        requestGet.onFirstCall().yields(null, { statusCode: 200 }, {});
+      function (done) {
+        requestPost.onFirstCall().yields(null, {statusCode: 200}, JSON.stringify(successfulAuthResponse));
+        requestGet.onFirstCall().yields(null, {statusCode: 200}, {});
 
         requestWithOAuth2.get('http://testhost.com', 'http://oauthserver.com', 'clientId', 'clientSecret', 'username',
-          'password', function(err) {
-            if (!err) {
-              request.post.should.have.been.calledWith(expectedOAuthRequest);
-              request.get.should.have.been.calledWith(expectedTesthostRequest);
-            }
-            done(err);
+          'password', function (err, res, body) {
+            request.post.should.have.been.calledWith(expectedOAuthRequest);
+            request.get.should.have.been.calledWith(expectedTesthostRequest);
+            done();
           });
       });
 
-    it('Uses an existing authorization token if present', function(done) {
-      // First call to oAuth server
-      requestPost.yields(null, { statusCode: 200 }, JSON.stringify(successfulAuthResponse));
-      // First call to URL
-      requestGet.yields(null, { statusCode: 200 }, 'Woo');
-      requestWithOAuth2.get('http://testhost.com', 'http://oauthserver.com',
-        'clientId', 'clientSecret', 'username', 'password', function(err) {
-        if (!err) {
-          request.post.should.have.been.calledWith(expectedOAuthRequest);
-          request.get.should.have.been.calledWith(expectedTesthostRequest);
-          requestWithOAuth2.get('http://testhost.com', 'http://oauthserver.com', 'clientId',
-            'clientSecret', 'username', 'password', function(err2) {
-            // Check it doesn't authorize with multiple requests
-            if (!err2) {
-              assert(request.post.calledOnce);
-            }
-            done(err2);
-          });
-        } else {
-          done(err);
-        }
+    it('Uses an existing authorization token if present', function (done) {
```

Comment:
I think these changes are ok, but would like to understand what issues Robin was trying to solve by handling the error explicitly in the test